### DISCUSSION TITLE
`odisAuthError` due to incorrect Data Encryption Key (DEK)

### DISCUSSION THREAD

**Discussion by vibern0** [celo-org/social-connect#209] [Help (Q&A)] [answered]
I've followed the implementation and had it working a few weeks back.
Picked up now to do some clean up and prepare to make it live and I'm getting "odisAuthError" on calls like getObfuscatedIdentifier or getPnpQuotaStatus.
What could it be?

**Comment by arthurgousset** [comment, marked as answer]
For context @therealharpaljadeja: @obernardovieira is using "@celo/identity": "3.2.0".

Could you share more context on your implementation please: 
- what `authSigner` settings are you using (DEK vs issuer key), e.g.

```js
authSigner: AuthSigner = {
    authenticationMethod: AuthenticationMethod.ENCRYPTION_KEY,
    rawKey: DEK_PRIVATE_KEY,
  };
```

- what client libraries are you using? (e.g. ethers, web3, contractkit)

**Reply by vibern0** [reply]
Sorry, sure.
https://github.com/impactMarket/backend/blob/main/packages/api/src/services/attestation/index.ts#L92

The DEK is this account 0x10eD22C400d8089f9f4422E4083Cd4ad446dDd2C and the issuer 0xe3475047EF9F9231CD6fAe02B3cBc5148E8eB2c8

**Reply by vibern0** [reply]
ahh wait 🤦 using wrong DEK 🤦 🤦 🤦 🤦 sorry. Nevermind, it's working.

**Reply by therealharpaljadeja** [reply]
Considering this Resolved! 😅

**Reply by Ronex-Ondimu** [reply]
whats the correct DEK and Issuer?

**Comment by arthurgousset** [comment]
(In case it helps tracing)

According to the ODIS source code, the "odisAuthError" is thrown when there is a 401 error. 

```js
export enum ErrorMessages {
   // ...
  ODIS_AUTH_ERROR = 'odisAuthError',
  // ...  
}

// ...

switch (res.status) {
   // ...
  case 401:
    throw new Error(ErrorMessages.ODIS_AUTH_ERROR)
  // ...
}
```

Source: [Github > celo-org > monorepo > packages > sdk > identity > src > odis > query.ts](https://github.com/celo-org/celo-monorepo/blob/47c9ebb6b0b64795625be654502e8134edcdacf8/packages/sdk/identity/src/odis/query.ts#L206-L207)

**Comment by Ronex-Ondimu** [comment]
how do someone fix the 401 error?
